grammar.ebnf

```diff
@@ -33,14 +42,54 @@ Statement = "{", {Statement}, "}"
   because we can achieve that as:
   Arr arr = a.b;  // arr is a pointer under the hood - any change to its
   elements will be reflected to a.b arr = a.b;
-*/
+*)
+
 LValue = Identifier, [ ".", Identifier ];
 
+Expression = FunctionCall
+           | NotExpression
+           | ArrayLookupExpression
+           | BinaryExpression;
+
+FunctionCall = Identifier, "(", FuncCallArgList, ")";
+
 FuncCallArgList = [ Expression, {",", Expression} ];
 
+(*
+  Expression precedence is simulated by using levels.
+  The higher the level, the bigger the precedence.
+*)
+
+BinaryExpression = Level1Expr, [ "&&", Level1Expr, [BinaryExpression]];
```

Comment:
You don't need the `[BinaryExpression]`.  You can just write `BinaryExpression = Level1Expr, [ "&&", Level1Expr]`.
Also, `&&` should have higher precedence than `||`.
Lastly, you can name those in a more descriptive way, like `Level1Expr -> OrExpr`, `Level2Expr -> CmpExpr` etc.

Reply:
You mean, `BinaryExpression = Level1Expr, { "&&", Level1Expr}` right?

Reply:
Yes thanks, my bad, I meant to use repetition.